src/core/components/keep-alive.js

```diff
@@ -85,8 +85,8 @@ export default {
     const componentOptions: ?VNodeComponentOptions = vnode && vnode.componentOptions
     if (componentOptions) {
       // check pattern
-      const name: ?string = getComponentName(componentOptions)
-      if (name && (
+      const name: ?string = componentOptions && componentOptions.Ctor.options.name
```

Comment:
why do you change the way of getting name? Plus doesn't need `componentOptions &&`  since it's already in the if branch.
IMO it should keep original logic ` opts.Ctor.options.name || opts.tag`

Reply:
Thanks for review , already modified.

docs descripition:
The match is first checked on the component’s own name option, then its local registration name (the key in the parent’s components option) if the name option is not available. Anonymous components cannot be matched against.

Anonymous components always have a registered name(  tag in code) , i think the descripition of docs may cause logical hole.

Reply:
@cuteyumiko IMO we may need to keep `opts.tag`. For example:
``` js
const foo = {
  template: `<div>foo</div>`
}

new Vue({
  el: '#app',
  components: {
    foo
  }
})
``` 

In above case, you'll find `Ctor.options` doesn't has a `name` property, but there actually is `tag` in it.

Reply:
ehhh..i see 
i have done another test for caching anonymous components , it turns out caching logic is ok

 #6938  it's problem may be casued by vue-router , i tracked this issue in a wrong way